src/tools/debugger/mhpdbgc.c

```diff
@@ -546,58 +555,57 @@ static void mhp_stop(int argc, char * argv[])
    }
 }
 
-static void mhp_trace(int argc, char * argv[])
+static void mhp_trace(int argc, char *argv[])
 {
-   if (!mhpdbgc.stopped) {
-      mhp_printf("must be in stopped state\n");
-   } else {
-      mhpdbgc.stopped = 0;
-      if (dpmi_active())
-        dpmi_mhp_setTF(1);
-      set_TF();
+  if (!check_for_stopped())
+    return;
 
-      if (!strcmp(argv[0], "ti")) {
-	mhpdbgc.trapcmd = 1;
-      } else {
-	mhpdbgc.trapcmd = 2;
-      }
+  mhpdbgc.stopped = 0;
+  if (dpmi_active())
+    dpmi_mhp_setTF(1);
+  set_TF();
 
-      mhpdbgc.trapip = mhp_getcsip_value();
-
-      if (!in_dpmi_pm()) {
-	unsigned char *csp = SEG_ADR((unsigned char *), cs, ip);
-	switch (csp[0]) {
-	case 0xcd:
-	    if (mhpdbgc.trapcmd != 1)
-		break;
-	    LWORD(eip) += 2;
-	    do_int(csp[1]);
-	    set_TF();
-	    mhpdbgc.stopped = 1;
-	    mhpdbgc.int_handled = 1;
-	    mhp_cmd("r0");
-	    break;
-	case 0xcf:
-	    LWORD(eip) += 1;
-	    fake_iret();
-	    set_TF();
-	    mhpdbgc.stopped = 1;
-	    mhp_cmd("r0");
-	    break;
-	}
-      }
-   }
+  if (!strcmp(argv[0], "ti")) {
+    mhpdbgc.trapcmd = 1;
+  } else {
+    mhpdbgc.trapcmd = 2;
+  }
+
+  mhpdbgc.trapip = mhp_getcsip_value();
+
+  if (!in_dpmi_pm()) {
+    unsigned char *csp = SEG_ADR((unsigned char *), cs, ip);
+    switch (csp[0]) {
+      case 0xcd:
+        if (mhpdbgc.trapcmd != 1)
```

Comment:
Here's where you can put the new
`int` handling that is discussed in another
thread, btw.

Reply:
Thanks, I'll have a look at that now.